src/main/java/mezz/jei/plugins/vanilla/ingredients/EnchantDataRenderer.java

```diff
@@ -0,0 +1,83 @@
+package mezz.jei.plugins.vanilla.ingredients;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+import mezz.jei.api.ingredients.IIngredientRenderer;
+import mezz.jei.util.ErrorUtil;
+import mezz.jei.util.Log;
+import mezz.jei.util.Translator;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.enchantment.EnchantmentData;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Items;
+import net.minecraft.item.EnumRarity;
+import net.minecraft.item.ItemEnchantedBook;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.TextFormatting;
+
+public class EnchantDataRenderer implements IIngredientRenderer<EnchantmentData> {
```

Comment:
You should pass in an instance of `IIngredientRenderer<ItemStack>` to the constructor of this renderer, and use it to render the enchanted books.